contracts/Treasury.sol

```diff
@@ -0,0 +1,173 @@
+pragma solidity ^0.6.5;
+
+import "@openzeppelin/contracts-upgradeable/proxy/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "abdk-libraries-solidity/ABDKMath64x64.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "./cryptoblades.sol";
+
+contract Treasury is Initializable, AccessControlUpgradeable {
+    using SafeMath for uint256;
+    using ABDKMath64x64 for int128;
+    using SafeERC20 for IERC20;
+
+    bytes32 public constant GAME_ADMIN = keccak256("GAME_ADMIN");
+
+    uint256 public multiplierUnit;
+
+    PartnerProject[] private partneredProjects;
+    mapping(uint256 => uint256) projectAddedBlockNumber;
+    mapping(uint256 => uint256) tokensClaimed;
+    uint256 skillPrice;
```

Comment:
updating skill price in this contract will require a bot, effectively making this field an oracle of its own.
Is this done to save gas costs on making an external contract call so that the claim costs less for the user?
(it seems pretty good tbh - a bot update costs us 10 cents each atm, and we only need one every few hours)
Perhaps we expect to have slightly different values here than we'd have in a regular oracle?

Reply:
Actually I wasn't thinking about gas saving aspect, but it makes sense. I was thinking that we need a seperate, more frequent updates here and I remember that the original oracle was usually lagging back in the days (not sure if that was intended to work that way or what was going on there).